src/core/types/ui_settings.ts

```diff
@@ -73,10 +72,22 @@ export interface UiSettingsParams {
   /*
    * Allows defining a custom validation applicable to value change on the client.
    * @deprecated
+   * Use schema instead.
    */
   validation?: ImageValidation | StringValidation;
+  /*
+   * Value validation schema
+   * Used to validate value on write and read.
+   */
+  schema: Type<T>;
 }
 
+/**
+ * A sub-set of {@link UiSettingsParams} exposed to the client-side.
+ * @public
+ * */
+export type PublicUiSettingsParams = Omit<UiSettingsParams, 'schema'>;
```

Comment:
I'm open to other naming suggestions. ther problem that https://github.com/elastic/kibana/pull/59694/files#diff-e3ce52429da99a601c0f3a737a9039daR51 is not an internal type

Reply:
Fine with me